7_Classes/class_example.py

```diff
@@ -0,0 +1,114 @@
+class Hero:
+    def __init__(self, hit_points, mana, strength, mind, weapon, race):
+        self.__hp = hit_points  # private, more secure than protected: accessible only in its class
+        self.__mana = mana
+        self._strength = strength  # protected; accessible only in their class and sub-classes (pass in inheritance)
+        self._mind = mind
+        self.weapon = weapon
+        self.race = race  # instance attributes
+
+    def display_visible_hero_info(self):  # public
+        print(f'Hero is {self.race} and has a {self.weapon}.')
+
+    def _display_base_stats(self):  # private
+        print(f'Strength: {self._strength}')
+        print(f'Mind: {self._mind}')
+
+    def __display_hp_mana_bars_values(self):  # protected
+        return {'HP': self.__hp, 'Mana': self.__mana}
+
+    def access_hp_mana(self):
+        return self.__display_hp_mana_bars_values()
+
+
+class Warrior(Hero):
+    _attack_type = 'Poison'  # class attribute
+
+    def __init__(self, hit_points, mind, mana, strength, weapon, race):
+        super().__init__(hit_points, mind, mana, strength,
+                         weapon, race)
```

Comment:
no need to call `super().__init__` if your class is initialized the same as the parent class.